### DISCUSSION TITLE
SQLite too old message on voidlinux

### DISCUSSION THREAD

**Discussion by RudiOnTheAir** [drakkan/sftpgo#1477] [General]
Hi

After upgrading the packages on voidlinux, sftpgo stopped working with an sqlite schema 19 too old, and the initprovider command seems to upgrade, but in the log the message stays at v19 error.

Any hints how to fix?

Thanks in advance

**Comment by drakkan** [comment]
The Void developers did not update the package for a long time and went from version 2.3.x to 2.5.x, this requires manual intervention, see the [upgrading docs](https://github.com/drakkan/sftpgo?tab=readme-ov-file#upgrading)

**Reply by RudiOnTheAir** [reply]
OK, but what can i do? I cannot install the 2.4., because that package isn't available? Should i switch over to another distribution like debian??

**Comment by drakkan** [comment]
> OK, but what can i do? I cannot install the 2.4., because that package isn't available? Should i switch over to another distribution like debian??

You should report the problem to Void, so they can provide a way to resolve it. In the short term just download a 2.4.x binary from the release page (it should work on Void, I'm not familiar with this distro) and update your database, then use the 2.5.x package provided by the distribution.

**Reply by RudiOnTheAir** [reply]
That has worked. But all users and settings are lost. Not really a big problem, but not save that it happens again..

Tnx

**Comment by drakkan** [comment]
> That has worked. But all users and settings are lost. Not really a big problem, but not save that it happens again..
> 

if you perform the update procedure correctly, nothing will be lost

**Reply by RudiOnTheAir** [reply]
I am convinced of that. But I wasn't aware of the problem with the version jump in advance. I think it might (for me) be better to build sftpgo by myself and then decide when i want to update it. Independent of the package manager.

**Comment by RudiOnTheAir** [comment]
Wait. The v19 db is to old message stays there... Maybe there is a problem with the locations of some files.? On Void, this is the startscript, bec. there is no systemd.!

`bash-5.2# cat run
#!/bin/sh
exec 2>&1

export SFTPGO_HTTPD__TEMPLATES_PATH=/usr/share/sftpgo/templates
export SFTPGO_HTTPD__STATIC_FILES_PATH=/usr/share/sftpgo/static
export SFTPGO_DATA_PROVIDER__DRIVER=sqlite
export SFTPGO_DATA_PROVIDER__NAME=/var/lib/sftpgo/sftpgo.db
export SFTPGO_SFTPD__HOST_KEYS=/var/lib/sftpgo/id_rsa,/var/lib/sftpgo/id_ecdsa,/var/lib/sftpgo/id_ed25519

[ -r ./conf ] && . ./conf

exec sftpgo serve
`

**Comment by RudiOnTheAir** [comment]
Yes, i could fix it. I think, without theese export lines the  2.4. release created a new db. So i renamed the binaery in /usr/bin/ and copied the old version on that place, so i was able to use the void startscript running only sv stop sftpgo and start two times. (with the old and the new bin...version @ /usr/bin/)

And the next time, void will update sftpgo i take a look at the release number..

Again tnx and mary xmas and a happy new year!
